restore information about truncated responses to sql api (#16001)
apache/druid: docs/api-reference/sql-api.md

* `query`: SQL query string.

* `resultFormat`: String that indicates the format to return query results. Select one of the following formats:
  * `object`: Returns a JSON array of JSON objects with the HTTP response header `Content-Type: application/json`.  

>      Each object's field names match the columns returned by the SQL query, and are provided in the same order as the SQL query.

Author: ```suggestion
     Object field names match the columns returned by the SQL query in the same order as the SQL query.
```

  * `array`: Returns a JSON array of JSON arrays with the HTTP response header `Content-Type: application/json`.  
     Each inner array has elements matching the columns returned by the SQL query, in order.

  * `objectLines`: Returns newline-delimited JSON objects with the HTTP response header `Content-Type: text/plain`.  
     Newline separation can make it easier to parse the entire response set as a stream if you don't have a streaming JSON parser.

>      To make it possible to detect a truncated response, this format includes a trailer of one blank newline character.

Author: ```suggestion
     This format includes a trailer of one blank newline character so you can detect a truncated response.
```

  * `arrayLines`: Returns newline-delimited JSON arrays with the HTTP response header `Content-Type: text/plain`.  

>      Newline separation can make it easier to parse the entire response set as a stream if you don't have a streaming JSON parser.

Author: ```suggestion
     Newline separation facilitates parsing the entire response set as a stream if you don't have a streaming JSON parser.
```

>  This format includes a trailer of one blank newline character so you can detect a truncated response.

Author: ```suggestion
 This format includes a single trailing newline character so you can detect a truncated response.
```

>      To make it possible to detect a truncated response, this format includes a trailer of one blank newline character.

Author: ```suggestion
 This format includes a trailer of one blank newline character so you can detect a truncated response.
```

  * `csv`: Returns comma-separated values with one row per line. Sent with the HTTP response header `Content-Type: text/csv`.  
     Druid uses double quotes to escape individual field values. For example a value with a comma returns `"A,B"`.
     If the field value contains a double quote character, Druid escapes it with a second double quote character.
     For example `foo"bar` becomes `foo""bar`.

>       This format includes a trailer of one blank newline character so you can detect a truncated response.

Author: ```suggestion
      This format includes a single trailing newline character so you can detect a truncated response.
```

>      To make it possible to detect a truncated response, this format includes a trailer of one blank newline character.

Author: ```suggestion
      This format includes a trailer of one blank newline character so you can detect a truncated response.
```

* `resultFormat`: String that indicates the format to return query results. Select one of the following formats:
  * `object`: Returns a JSON array of JSON objects with the HTTP response header `Content-Type: application/json`.  
     Object field names match the columns returned by the SQL query in the same order as the SQL query.

  * `array`: Returns a JSON array of JSON arrays with the HTTP response header `Content-Type: application/json`.  
     Each inner array has elements matching the columns returned by the SQL query, in order.

  * `objectLines`: Returns newline-delimited JSON objects with the HTTP response header `Content-Type: text/plain`.  
     Newline separation facilitates parsing the entire response set as a stream if you don't have a streaming JSON parser.
     This format includes a single trailing newline character so you can detect a truncated response.

  * `arrayLines`: Returns newline-delimited JSON arrays with the HTTP response header `Content-Type: text/plain`.  
     Newline separation facilitates parsing the entire response set as a stream if you don't have a streaming JSON parser.

>  This format includes a single trailing newline character so you can detect a truncated response.

Reviewer: ```suggestion
     This format includes a single trailing newline character so you can detect a truncated response.
```
